File: 1-two-sum/1-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int[] answer = new int[2];
            for (int i = 0; i < nums.Length; i++) {
                for (int j = i + 1; j < nums.Length; j++){
                    if ((nums[i] + nums[j]) == target) {
                        answer[0] = i;
                        answer[1] = j;
                        return answer;
                    }
                }
            }
        return answer;
    }
}

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var map = {};
    
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] in map) {
            return [map[nums[i]], i];
        } else {
            let potentialMatch = target - nums[i];
            if (!(potentialMatch in map)) map[potentialMatch] = i;
        }
    }
}; 